### PR TITLE
{vis} [intel/2016b] Mesa/12.0.1 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**
Needs ~~#3440~~ and ~~#3407~~

### CODE REVIEW

**Comment by wpoely86** [comment]
Python is still a major issue here: currently we build it with Tk which needs X11... This needs to change.

File: easybuild/easyconfigs/t/Tk/Tk-8.6.4-intel-2016a-no-X11.eb [hunk 1]
```diff
@@ -20,6 +20,7 @@ dependencies = [
     ('zlib', '1.2.8'),
 ]
 
+# To be clear: this will still require the X11 to be present.
```
Comment (boegel) [inline easybuild/easyconfigs/t/Tk/Tk-8.6.4-intel-2016a-no-X11.eb]: if you add this, include a pointer to https://github.com/hpcugent/easybuild-easyconfigs/issues/2261

File: easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb [hunk 2]
```diff
@@ -18,7 +18,8 @@ dependencies = [
     ('libreadline', '6.3'),
     ('ncurses', '6.0'),
     ('SQLite', '3.13.0'),
-    ('Tk', '8.6.5'),
+    # We don't build Tkinter as it requires an X11 stack
+    # ('Tk', '8.6.5'),
```
Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb]: @wpoely86 this should be done in a separate PR for sure, together with providing a separate easyconfig for `Tkinter` itself

also, we shouldn't make this change before EasyBuild v3.0.0

Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb]: Then all the work is fully blocked. No Mesa, no openfoam, no paraview. This is a major blocker and it needs to go

File: easybuild/easyconfigs/l/libxml2/libxml2-2.9.4-intel-2016b-Python-2.7.12.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+easyblock = 'ConfigureMake'
```
Comment (boegel) [inline easybuild/easyconfigs/l/libxml2/libxml2-2.9.4-intel-2016b-Python-2.7.12.eb]: @wpoely86 for libxml2 with Python bindings, please use the existing `libxml2` easyblock

We can enhance that easyblock so it can be used for *all* `libxml2` easyconfigs, to resolve the split-brain situation we have now where `libxml2` without Python is built via `ConfigureMake`

File: easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb]: @wpoely86 only if you don't have X11 in the OS, but we do...

Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb]: then you need to put in os deps. This PR doesn't work without this line.

Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb]: it never works, or only if you don't have X11 on the OS?

Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb]: you need a X11 stack for Tk.

Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb]: I know you do, but I still don't see how that's a blocker for this PR.

Currently, our Tk easyconfigs require to have X11 provided by the OS, whatever the versionsuffix or configure options say. We should definitely fix that, but this shouldn't block any other PRs?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2407.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/ad8bc5e3c76fb5b8241e9d525092cb3d for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 9 out of 9 (8 easyconfigs in this PR)
node2007.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2ca92997e6328d563542cd999fd7390d for a full test report.

File: easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb [hunk 4]
```diff
@@ -18,6 +18,7 @@ dependencies = [
     ('libreadline', '6.3'),
     ('ncurses', '6.0'),
     ('SQLite', '3.13.0'),
+    # Be ware that this requires a full X11 stack
```
Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb]: pfft, is this worth adding here? we already have open issues for this, etc.

Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.12-intel-2016b.eb]: yes, people will look at the easyconfigs but not that the issues. And this is major.

File: easybuild/easyconfigs/t/Tk/Tk-8.6.4-intel-2016a-no-X11.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/t/Tk/Tk-8.6.4-intel-2016a-no-X11.eb]: also, this is totally unrelated to this PR (this particular Tk easyconfig isn't used here, it's even a different toolchain)

let's not start adding comments left and right for stuff we already have open issues for

**Comment by boegel** [comment]
lgtm

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 6 out of 9 (8 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/462139070efa3b545ca21076a29825e8 for a full test report.

**Comment by wpoely86** [comment]
The last one shows what happens on systems without X.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/55f82f7bb1da452785923dde8ec78755 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @wpoely86!
